### PR TITLE
docs: cloud docs content and better navigation from index

### PR DESCRIPTION
**Pull request by pnadolny13**
Related to https://github.com/meltano/meltano/issues/7709 and https://github.com/meltano/meltano/issues/7677

Prior art:
- Dagster https://docs.dagster.io/dagster-cloud
- Airbyte https://docs.airbyte.com/cloud/getting-started-with-airbyte-cloud and https://docs.airbyte.com/category/using-airbyte-cloud

I wanted to add some better hints on the index page so users know where to go next, currently its hard to navigate:
- Adds really basic marketing content and links to the main site
- Remove index list
- Add getting started section to index. What do we think about this recommendation to init a project then add a sample schedule? I think we'd prefer to get people who are new to meltano onboarded first before they start diving into configuring their actual taps/targets.
- Organize links to other docs to make them easier to find
- Removes encrypting secrets page because its not really relevant anymore now that we have a config cli section. Also calling it "encrypting" is a bit confusing and I think was just left over from the old process. In the process I moved the reserved variables warning to the CLI reference.
- Added a few sections in a usage page, still TBD: backfills and state, cloud UI, and managing credits.

I don't think this is the best and final structure for these docs but I think its easier to navigate than what we had before. I can imagine us having a CLI reference along with detailed usage sections for the major commands.

cc @tayloramurphy @sbalnojan what do you think?

### CODE REVIEW

**Review by tayloramurphy — APPROVED** [review]
@pnadolny13 small change but I think this is worth deploying now

**Comment by pnadolny13** [comment]
@tayloramurphy this should be good to go! I finished up the placeholder sections I had, move the getting started steps to a standalone page, added the known limitations page to the side bar, and reordered the side bar based on importance.

**Comment by pnadolny13** [comment]
@WillDaSilva @magreenbaum can you confirm for me that the known limitations page is accurate here?

File: docs/src/_cloud/known_issues.md [hunk 1]
```diff
@@ -20,6 +21,16 @@ In the future, Meltano Cloud will automatically run any schedules you have confi
...
+### Manual Deployment Of All Changes
+
+Meltano cloud will have support for automatic deployments in GA but currently users need to manually request a re-deployment of a project when new commits are added to the git repository.
+
+```bash
+meltano-cloud deployment update --name prod
+```
+
+See the [deployments CLI reference](/cloud/cloud-cli#deployment).
+
```
Comment (magreenbaum) [inline docs/src/_cloud/known_issues.md:33]:
```suggestion

### Self-Managed Addition of Projects

Meltano cloud will have support to add and manage projects in the near future but currently a project addition needs to be requested through Meltano Cloud Support.

See related issue: [#7412](https://github.com/meltano/meltano/issues/7412)

```

**Review by magreenbaum — COMMENTED** [review]
Might be worth mentioning we don't currently support self-managed project additions. Other than that, known limitations page lgtm.

**Review by magreenbaum — APPROVED** [review]

**Review by tayloramurphy — APPROVED** [review]

**Pull request merged**